src/python/grpcio_tests/tests/channelz/_channelz_page_test.py

```diff
@@ -0,0 +1,212 @@
+        try:
+            self._channel.unary_unary(_FAILED_UNARY_UNARY)(_REQUEST)
+        except grpc.RpcError:
+            return
+
+    def _send_successful_stream_stream(self):
+        response_iterator = self._channel.stream_stream(
+            _SUCCESSFUL_STREAM_STREAM).__call__(
+                iter([_REQUEST] * test_constants.STREAM_LENGTH))
+        cnt = 0
+        for _ in response_iterator:
+            cnt += 1
+        self.assertEqual(cnt, test_constants.STREAM_LENGTH)
+
+    def setUp(self):
+        super(ChannelzPageTest, self).setUp()
+        self._server = test_common.test_server()
+        port = self._server.add_insecure_port('[::]:0')
+        self._server.add_generic_rpc_handlers((_GenericHandler(),))
+        self._server.start()
+
+        self._channel = grpc.insecure_channel('localhost:%d' % port)
+
+        self._page_server = channelz.serve_channelz_page('', 0)
+        self._page_url_prefix = _CHANNELZ_URL_PREFIX_TEMPLATE % self._page_server.server_address[
+            1]
+
+        # Emit RPCs to make sure sockets are created
+        self._send_successful_unary_unary()
+        self._send_failed_unary_unary()
+        self._send_successful_stream_stream()
+
+    def tearDown(self):
+        self._page_server.shutdown()
+        self._page_server.server_close()
+        self._server.stop(None)
+        self._channel.close()
+        super(ChannelzPageTest, self).tearDown()
+
+    def test_homepage(self):
+        resp = requests.get(self._page_url_prefix)
+        self.assertEqual(resp.status_code, 200)
+
+    def test_channel(self):
+        # Page of list of channels
+        resp = requests.get(self._page_url_prefix + 'topchannels')
+        self.assertEqual(resp.status_code, 200)
+
+        # Page of detail of a channel
+        surffix = _table_a_href_re.search(resp.text).group(1)
+        self.assertIn('channel?channel_id=', surffix)
+        resp = requests.get(self._page_url_prefix + surffix)
+        self.assertEqual(resp.status_code, 200)
+
+        # Page of detail of a subchannel
+        surffix = _table_eq1_a_href_re.search(resp.text).group(1)
+        # surffix = pq(resp.text)('table').eq(1).find('a').attr('href')
+        self.assertIn('subchannel?subchannel_id=', surffix)
+        resp = requests.get(self._page_url_prefix + surffix)
+        self.assertEqual(resp.status_code, 200)
+
+        # Page of detail of a socket
+        surffix = _table_a_href_re.search(resp.text).group(1)
+        self.assertIn('socket?socket_id=', surffix)
+        resp = requests.get(self._page_url_prefix + surffix)
+        self.assertEqual(resp.status_code, 200)
+
+    def test_serversockets(self):
+        # Page of list of servers
+        resp = requests.get(self._page_url_prefix + 'servers')
+        self.assertEqual(resp.status_code, 200)
+
+        # TODO(lidiz) In Python 3, the server from last test unit will remain
+        # alive... Remove the selection logic when the deallocation is fixed.
```

Comment:
As discussed offline, this comment indicates either a bug in Python or a bug in core's channelz implementation; we should identify which is the case.